packages/lib/src/project/BaseSimpleProject.ts

```diff
@@ -0,0 +1,148 @@
+import isEmpty from 'lodash.isempty';
+
+import Proxy from '../proxy/Proxy';
+import Logger from '../utils/Logger';
+import Package from '../application/Package';
+import { deploy } from '../utils/Transactions';
+import { toAddress } from '../utils/Addresses';
+import { bytecodeDigest } from '../utils/Bytecode';
+import { toSemanticVersion }from '../utils/Semver';
+import { buildCallData, callDescription, CalldataInfo } from '../utils/ABIs';
+import { ContractInterface } from './AppProject';
+import ContractFactory, { ContractWrapper } from '../artifacts/ContractFactory';
+
+const log: Logger = new Logger('BaseSimpleProject');
+
+interface Implementations {
+  [contractName: string]: Implementation;
+}
+
+interface Implementation {
+  address: string;
+  bytecodeHash: string;
+}
+
+interface Dependencies {
+  [packageName: string]: Dependency;
+}
+
+interface Dependency {
+  package: string;
+  version: string;
+}
+
+export default abstract class BaseSimpleProject {
```

Comment:
If there is no other kind of Base Projects, I would call this one `BaseProject` directly

Reply:
Yeap, there is another Base Project. We have two main structures: `BasePackageProject` - `PackageProject` - `AppProject` and `BaseSimpleProject` - `SimpleProject` - `ProxyAdminProject`, the former in charge of managing published projects and the latter in charge of managing non-published ones. Hopefully, in the near future we'll be able to reduce at least this two structures by merging some classes and finally get to the point where only an `AppProject` and a `ProxyAdminProject` (maybe named `PublishedProject` and `UnpublishedProject`?) will be necessary.